File: atcoder/abc363/c.rs
#![allow(unused)]

fn main() {
    let inp = readv::<usize>();
    let (n, k) = (inp[0], inp[1]);
    let mut s = reads();
    s.sort();

    let mut ans = 0;
    loop {
        let any = s.windows(k).any(|w| (0..k).all(|i| w[i] == w[k - 1 - i]));
        if !any {
            ans += 1;
        }
        
        if next_permutation(&mut s).is_none() {
            break;
        }
    }

    println!("{}", ans);
}

fn next_permutation<T: Ord>(arr: &mut [T]) -> Option<()> {
    let k = arr.windows(2).rposition(|w| w[0] < w[1])?;
    let j = arr.iter().rposition(|a| a > &arr[k]).unwrap();
    arr.swap(k, j);
    arr[(k + 1)..].reverse();
    Some(())
}

fn read<T: std::str::FromStr>() -> T {
    let mut s = String::new();
    std::io::stdin().read_line(&mut s).ok();
    s.trim().parse().ok().unwrap()
}

fn readv<T: std::str::FromStr>() -> Vec<T> {
    read::<String>()
        .split_ascii_whitespace()
        .map(|t| t.parse().ok().unwrap())
        .collect()
}

fn reads() -> Vec<char> {
    read::<String>().chars().collect()
}

fn mapv<T, S, F: Fn(&T) -> S>(arr: &Vec<T>, f: F) -> Vec<S> {
    arr.iter().map(f).collect()
}

fn join<T: ToString>(arr: &[T], sep: &str) -> String {
    arr.iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join(sep)
}

File: atcoder/abc363/c.py
def next_permutation_(arr):
    n = len(arr)
    k = next(i for i in reversed(range(n - 1)) if arr[i] < arr[i + 1])
    j = next(i for i in reversed(range(n)) if arr[i] > arr[k])
    arr[k], arr[j] = arr[j], arr[k]
    arr[(k + 1) :] = arr[(k + 1) :][::-1]


N, K = map(int, input().split())
S = list(input())
S.sort()

ans = 0
while True:
    has_palindrome = False
    for i in range(N - K + 1):
        if all(S[i + j] == S[i + K - 1 - j] for j in range(K)):
            has_palindrome = True
            break
    if not has_palindrome:
        ans += 1

    try:
        next_permutation_(S)
    except StopIteration:
        break

print(ans)
